### DISCUSSION TITLE
Universal Display Driver SH1106 with SPI

### DISCUSSION THREAD

**Discussion by markkrz** [arendst/Tasmota#12880] [Q&A] [answered]
Hi I've been trying to use the Universal Display Driver with an SPI version of the SH1106 OLED display connected to a Wemos D1Mini. 

CLK on GPIO14, MOSI GPIO13, SPI DC GPIO4, SPI CS GPIO15, Reset GPIO5 and Option A3 GPIO16. 

Have compiled the latest Tasmota Development version with UFILESYS, USE_DISPLAY, USE_UNIVERSAL_DISPLAY and USE_SPI added to user_config_override.h. 

I have modified the display.ini based on the SH1106 i2C examples, information on the Tasmota displays web page, datasheets and other SPI examples.  

Display.ini contents:

:H,SH1106,128,64,1,SPI,1,*,*,*,*,-1,*,-1,8
:S,0,2,1,0,30,20
:I
AE,0
D5,1,80
A8,1,3F
D3,1,00
40,0
8D,1,14
20,1,00
A1,0
C8,0
DA,1,12
81,1,CF
D9,1,F1
DB,1,40
A4,0
A6,0
AF,0
:o,AE
:O,AF
:A,00,10,40,00,02
:i,A6,A7
#
Just wondering if I've interpreted the display.ini configuration for an SPI display correctly? e.g for the Initial Register setup - I: the first value is the command, then the number of arguments and the the arguments itself. Where there is no argument, I've just entered 0 based on a few other examples. 

The i2C Display.ini version:

:H,SH1106,128,64,1,I2C,3c,*,*,*
:S,0,2,1,0,30,20
:I
AE
D5,80
A8,3f
D3,00
40
8D,14
20,00
A1
C8
DA,12
81,CF
D9,F1
DB,40
A4
A6
AF
:o,AE
:O,AF
:A,00,10,40,00,02
:i,A6,A7
#

The end result is that text is displayed on the screen, but it flashes on and off rapidly. In addition, not all commands work, for example displaytext [z] does not clear the screen. 

I've tried playing around with the Initialisation format, which stops the text flashing on and off, but other commands such as displaytext [z] still don't work. 

Any suggestions?

**Comment by ascillato2** [comment, marked as answer]
Since you haven't had any answer here, please, address this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4). Thanks.

---------------------------

### Support Information 
See [Docs](https://tasmota.github.io/docs/) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
See [Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)

**Comment by markkrz** [comment]
OK Thanks. Will do.
